### PR TITLE
add patch to fix buffer overflow in OpenMPI 4.1.x

### LINKED ISSUE #12978: Segmentation fault with FFTW/3.3.9/gompi-2021.04 on x86
I'm unable to install FFTW-3.3.9-gompi-2021.04.eb on a haswell node due to a segfault during the tests:

```
make[3]: Entering directory `/tmp/s3248973-EasyBuild/FFTW/3.3.9/gompi-2021.04/fftw-3.3.9/mpi'
perl -w ../tests/check.pl --verbose --random --maxsize=10000 -c=10  --mpi "mpirun -np 1 `pwd`/mpi-bench"
Executing "mpirun -np 1 /tmp/s3248973-EasyBuild/FFTW/3.3.9/gompi-2021.04/fftw-3.3.9/mpi/mpi-bench --verbose=1   --verify 'ok]4bx11hx10e00' --verify 'ik]4bx11hx10e00' --verify 'obr[48x18' --verify 'ibr[48x18' --verify 'obc[48x18' --verify 'ibc[48x18' --verify 'ofc[48x18' --verify 'ifc[48x18' --verify 'okd]5e11x2hx11e00x5o10v2' --verify 'ikd]5e11x2hx11e00x5o10v2' --verify 'okd11bx7e01x6h' --verify 'ikd11bx7e01x6h' --verify 'ofr]10x5x9x11' --verify 'ifr]10x5x9x11' --verify 'obc]10x5x9x11' --verify 'ibc]10x5x9x11' --verify 'ofc]10x5x9x11' --verify 'ifc]10x5x9x11' --verify 'obr[12x8x6x2' --verify 'ibr[12x8x6x2' --verify 'obc[12x8x6x2' --verify 'ibc[12x8x6x2' --verify 'ofc[12x8x6x2' --verify 'ifc[12x8x6x2' --verify 'ok]45hx16o00' --verify 'ik]45hx16o00' --verify 'obr2x7v2' --verify 'ibr2x7v2' --verify 'ofr2x7v2' --verify 'ifr2x7v2' --verify 'obc2x7v2' --verify 'ibc2x7v2' --verify 'ofc2x7v2' --verify 'ifc2x7v2' --verify 'obcd660' --verify 'ibcd660' --verify 'ofcd660' --verify 'ifcd660' --verify 'obr[6x12' --verify 'ibr[6x12' --verify 'obc[6x12' --verify 'ibc[6x12' --verify 'ofc[6x12' --verify 'ifc[6x12'"
[taurusa5:04118] *** Process received signal ***
[taurusa5:04118] Signal: Segmentation fault (11)
[taurusa5:04118] Signal code:  (128)
[taurusa5:04118] Failing at address: (nil)
[taurusa5:04118] [ 0] /lib64/libpthread.so.0(+0xf5f0)[0x2b72d76f15f0]
[taurusa5:04118] [ 1] /scratch/ws/1/s3248973-EasyBuild/easybuild-haswell/software/OpenMPI/4.1.1-GCC-10.3.0/lib/libmca_common_ofi.so.10(opal_mca_common_ofi_select_provider+0x1f0)[0x2b72d9982dd0]
[taurusa5:04118] [ 2] /scratch/ws/1/s3248973-EasyBuild/easybuild-haswell/software/OpenMPI/4.1.1-GCC-10.3.0/lib/openmpi/mca_btl_ofi.so(+0x3469)[0x2b72d9979469]
[taurusa5:04118] [ 3] /scratch/ws/1/s3248973-EasyBuild/easybuild-haswell/software/OpenMPI/4.1.1-GCC-10.3.0/lib/libopen-pal.so.40(mca_btl_base_select+0x109)[0x2b72d7dfb019]
[taurusa5:04118] [ 4] /scratch/ws/1/s3248973-EasyBuild/easybuild-haswell/software/OpenMPI/4.1.1-GCC-10.3.0/lib/openmpi/mca_bml_r2.so(mca_bml_r2_component_init+0x14)[0x2b72d9948764]
[taurusa5:04118] [ 5] /scratch/ws/1/s3248973-EasyBuild/easybuild-haswell/software/OpenMPI/4.1.1-GCC-10.3.0/lib/libmpi.so.40(mca_bml_base_init+0x81)[0x2b72d6fa7161]
[taurusa5:04118] [ 6] /scratch/ws/1/s3248973-EasyBuild/easybuild-haswell/software/OpenMPI/4.1.1-GCC-10.3.0/lib/libmpi.so.40(ompi_mpi_init+0x67b)[0x2b72d6fea71b]
[taurusa5:04118] [ 7] /scratch/ws/1/s3248973-EasyBuild/easybuild-haswell/software/OpenMPI/4.1.1-GCC-10.3.0/lib/libmpi.so.40(PMPI_Init_thread+0x57)[0x2b72d6f8b737]
[taurusa5:04118] [ 8] /tmp/s3248973-EasyBuild/FFTW/3.3.9/gompi-2021.04/fftw-3.3.9/mpi/.libs/mpi-bench[0x405815]
[taurusa5:04118] [ 9] /tmp/s3248973-EasyBuild/FFTW/3.3.9/gompi-2021.04/fftw-3.3.9/mpi/.libs/mpi-bench[0x406c14]
[taurusa5:04118] [10] /lib64/libc.so.6(__libc_start_main+0xf5)[0x2b72d7920505]
[taurusa5:04118] [11] /tmp/s3248973-EasyBuild/FFTW/3.3.9/gompi-2021.04/fftw-3.3.9/mpi/.libs/mpi-bench[0x402d3e]
[taurusa5:04118] *** End of error message ***
```

This is an E5-2680 v3 CPU on a node running RHEL 7.9. Rebuilding OpenMPI didn't help and the error happens always

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

fixes #12978, see also https://github.com/open-mpi/ompi/issues/9018

### CODE REVIEW

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb [hunk 1]
```diff
@@ -26,6 +29,8 @@ dependencies = [
     ('PMIx', '3.2.3'),
 ]
 
+toolchainopts = {'debug': True}
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb:32]: @Flamefire We don't always want to enable debug symbols, do we?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb:34]: My bad... Wasn't intentionally

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusi6333.taurus.hrsk.tu-dresden.de - Linux RHEL 7.9, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/6a6597f96408096111c5335295e6fcb6 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
taurusml25 - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 2.7.5
See https://gist.github.com/d6a85e4dd938caefa299ec2e12685a10 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/d7353f61261231e5d358c8878a42a85c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
easybuild2.novalocal - Linux centos linux 8.3.2011, POWER, IBM pSeries (emulated by qemu) (power9le), Python 3.6.8
See https://gist.github.com/0036f1245a100ab0d2e731c7e6a120b2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3101.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/4d190f36459b37b50e3e7814f2ccb5ce for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2618.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/ab010ad067fdefa88f704d604b4ca2c2 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3502.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/992a01754a12ee1c79591377db7a270c for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Flamefire!

**Pull request merged**